src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components;
+
+namespace Bit.Client.Web.BlazorUI
+{
+    public partial class BitNav
+    {
+        [Parameter] public string SelectedKey { get; set; }
+        [Parameter] public string AriaLabel { get; set; }
+        [Parameter] public bool IsOnTop { get; set; }
+        [Parameter] public List<NavLink> NavLinks { get; set; }
+        [Parameter] public EventCallback<NavLink> OnClick { get; set; }
+        [Parameter] public Func<NavLink, RenderFragment> OnRenderHeader { get; set; }
+        protected override string RootElementClass => "bit-nav";
+
+        protected virtual async Task Toggle(NavLink navLink)
+        {
+            if (IsEnabled is false) return;
+
+            if (navLink.Disabled) return;
+
+            if (navLink.Links?.Any() ?? false)
+            {
+                navLink.IsExpanded = !navLink.IsExpanded;
+                StateHasChanged();
```

Comment:
why calling "StateHasChanged"?
I think it is not required because this method called in the click event and the blazor itself will handle the dirty checks.